api/lagoon/client/usergroups.go

```
@@ -229,6 +229,35 @@ func (c *Client) UserCanSSHToEnvironment(
 	return c.client.Run(ctx, req, &environment)
 }
 
+// UserCanViewEnvironmentRoute queries the Lagoon API as a user to check if the user has access to
+// view the routes of an environment.
+func (c *Client) UserCanViewEnvironmentRoute(ctx context.Context, namespace string) (bool, error) {
+	req, err := c.newRequest("_lgraphql/usergroups/userCanViewEnvironmentRoute.graphql",
+		map[string]string{"namespace": namespace})
+	if err != nil {
+		return false, err
+	}
+
+	// Introduces a new struct that fits the response from userCanViewEnvironmentRoute.graphql 
+	type userCanViewEnvironmentRouteResponse struct {
+		UserCanViewEnvironmentRoute struct {
+			KubernetesNamespaceName string `json:"kubernetesNamespaceName"`
+		} `json:"userCanViewEnvironmentRoute"`
+	}
+
+	var resp userCanViewEnvironmentRouteResponse
+	err = c.client.Run(ctx, req, &resp)
+	if err != nil {
+		return false, err
+	}
+
+	if resp.UserCanViewEnvironmentRoute.KubernetesNamespaceName == "" {
+		return false, nil
+	}
+
+	return true, err
```

This response differs from what the API does itself. It should return what the API does.

If the API returns an error, this error should be returned to the user the same as others do. If the permission check fails, it will return an unauth error.

Error should = no access when you're consuming

---

thats what i did originally, but this and `UserCanSSHToEnvironment` API calls dont return errors when accessed by `guest@example.com`, for example (at least as far as I could get to work). Ill look into it more.

---

Hmm, maybe they don't error on the has permission then and just doesn't return a namespace if you have no permission

---

yeah, from memory that was exactly the behaviour i observed

---

https://github.com/uselagoon/lagoon/blob/8737fc6cf667d30a6b63fa3d5ef9e8840a581d13/services/api/src/resources/environment/resolvers.ts#L901-L910

Yeah that is exactly what it does :hide-the-pain.

So you can make your `userCanViewEnvironmentRoute` just fail the permission check and return that instead of `null` or you can reproduce what the ssh resolver does